kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/AbstractValidatingTextBox.java

```diff
@@ -85,16 +85,29 @@ public void onBlur(final BlurEvent event) {
                 String validationError = isValidValue(value,
                                                       true);
                 if (validationError != null) {
-                    fireValidationError(validationError);
-                    String validValue = makeValidValue(value);
-                    me.setValue(validValue);
-                    ValueChangeEvent.fire(AbstractValidatingTextBox.this,
-                                          validValue);
+                    fireValidation(value, validationError, me);
+                } else {
+                   validationError = isBalanced(value);
+                    if (validationError != null) {
+                        fireValidation(value, validationError, me);
+                    }
                 }
             }
         });
     }
 
+    private void fireValidation(String value, String validationError, TextBox me) {
+        fireValidationError(validationError);
+        String validValue = makeValidValue(value);
+        me.setValue(validValue);
+        ValueChangeEvent.fire(AbstractValidatingTextBox.this,
+                              validValue);
+    }
+
+    public String isBalanced(String string) {
```

Comment:
What is this method supposed to do? It return null everytime.

Reply:
Hi @domhanak the code works so that null is expected to be a valid value, if not returns the validationError description, this method returns null as it is the base class, actual implementation is in sub class.

Reply:
If I understand this correctly, a string isBalanced when the returned value is null? Imho that is bad design and will lead to NPE's in the future. I suggest changing it so that the method actually returns `true` or `false` if the string is balanced. Here returning true.

Reply:
Hi @domhanak I understand your point, but I just made it work just like isValidValue method from subclasses. Look at this implementation from CustomDataTypeBox.java, let me know what you think
![image](https://user-images.githubusercontent.com/39567766/143086405-fe728955-56d4-46b5-b245-a144cdd2419c.png)

Reply:
Ok, I ll wait for @Josephblt review

Reply:
I would make this method abstract and implement it on both CustomDataTypeTextBox and VariableNameTextBox.The same way isValidValue method does. 

I wouldn't worry about the true or false values. But I have to agree with @domhanak. The return types are confusing. Perhaps better naming of the methods to highlight their intent would improve things.

Reply:
Hi @Josephblt making it abstract and including GTLT in name to indicate balanced of Greater Than and Less Than